metagraph/src/graph/representation/canonical_dbg.cpp

```diff
@@ -165,79 +175,54 @@ ::call_outgoing_kmers(node_index node, const OutgoingEdgeCallback &callback) con
     size_t max_num_edges_left = children.size() - has_sentinel_;
 
     graph_->call_outgoing_kmers(node, [&](node_index next, char c) {
-        if (c != boss::BOSS::kSentinel) {
-            callback(next, c);
-            assert(traverse(node, c) == next);
-            children[alphabet_encoder_[c]] = next;
-            --max_num_edges_left;
-        }
+        assert(has_sentinel_ || c != boss::BOSS::kSentinel);
+        assert(c == boss::BOSS::kSentinel || traverse(node, c) == next);
+        callback(next, c);
+        children[alphabet_encoder_[c]] = next;
+        --max_num_edges_left;
```

Comment:
In some cases you call dummy k-mers, in some cases you skip them... Can you add comments to specify this?

Reply:
The idea here is to prevent a dummy sink/source k-mer from being called twice (one from each strand), since it could be that both orientations of a (k-1)-mer exist in the underlying graph.